pkg/watcher/results/results.go

```diff
@@ -118,7 +120,8 @@ func resultName(o metav1.Object) string {
 	return result.FormatName(o.GetNamespace(), part)
 }
 
-// upsertRecord updates or creates a record for the object
+// upsertRecord updates or creates a record for the object. If there has been
+// no change in the Record data, the existing Record is returned.
```

Comment:
This change seems unrelated to deleting completed runs, was this intentional?

Reply:
Nice, thanks for that explanation! Agree about putting this context into the PR description, and probably in a doc comment in here too.